### PR TITLE
Fix: Consent Engine `v1`

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Summary
Implement consent record creation with composite unique constraint and update OpenAPI documentation

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Refactoring (no functional changes)
- [x] Performance improvement
- [ ] Other (please describe):

## Changes Made
- Remove Complexity in the operations of CE
- Add logics to consent creations where there is a possibility for existence of another consent record

## Testing
- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues
Related to #337

### CODE REVIEW

File: exchange/consent-engine/v1/models/consent.go [hunk 1]
```diff
@@ -11,22 +11,23 @@ type ConsentRecord struct {
...
 	// AppName is the name of the consumer application
 	AppName *string `gorm:"column:app_name;type:varchar(255);" json:"app_name,omitempty"`
 	// Status is the status of the consent record: pending, approved, rejected, expired, revoked
 	Status string `gorm:"column:status;type:varchar(50);not null;index:idx_consent_records_status" json:"status"`
 	// Type is the type of consent mechanism "realtime" or "offline"
 	Type string `gorm:"column:type;type:varchar(50);not null" json:"type"`
 	// CreatedAt is the timestamp when the consent record was created
-	CreatedAt time.Time `gorm:"column:created_at;type:timestamp with time zone;not null;default:CURRENT_TIMESTAMP;index:idx_consent_records_created_at" json:"created_at"`
+	// Part of composite unique key (OwnerID, OwnerEmail, AppID, CreatedAt)
+	CreatedAt time.Time `gorm:"column:created_at;type:timestamp with time zone;not null;uniqueIndex:idx_consent_unique_tuple,composite:unique_tuple;default:CURRENT_TIMESTAMP;index:idx_consent_records_created_at" json:"created_at"`
```
Comment (ginaxu1) [inline exchange/consent-engine/v1/models/consent.go]: since created_at is a timestamp that defaults to CURRENT_TIMESTAMP, it is highly unique. 

nit: remove created_at from the generic unique constraint or replace it with a partial index ensuring only one pending/approved record exists per owner_id + app_id
